pkg/api/v1alpha1/tinkerbelltemplateconfig_defaults.go

```diff
@@ -74,14 +76,14 @@ func withNetplanAction(b v1alpha1.VersionsBundle) ActionOpt {
 	}
 }
 
-func withTinkCloudInitAction(b v1alpha1.VersionsBundle) ActionOpt {
+func withTinkCloudInitAction(b v1alpha1.VersionsBundle, diskType string) ActionOpt {
 	return func(a *[]tinkerbell.Action) {
 		*a = append(*a, tinkerbell.Action{
 			Name:    "add-tink-cloud-init-config",
 			Image:   b.Tinkerbell.TinkerbellStack.Actions.WriteFile.URI,
 			Timeout: 90,
 			Environment: map[string]string{
-				"DEST_DISK": "/dev/sda2",
+				"DEST_DISK": fmt.Sprintf("%s2", diskType),
```

Comment:
The partition number shouldn't be the responsibility of individual actions to define. They should just accept a destination disk, in full. 

The partition number, for now, could be defined by the templating construct perhaps? We could modify this in a subsequent PR as we explore supporting multiple disk types (NVMe etc).